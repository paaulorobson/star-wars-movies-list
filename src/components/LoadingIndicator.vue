<template>
  <div class="loading-indicator">
    <div class="loading-indicator__spinner">
      <svg xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 500 500" width="280">
          <title>LOADING</title>
          <path fill="transparent" d="M243.2, 382.4c-74.8,
          0-135.5-60.7-135.5-135.5s60.7-135.5,135.5-135.5s135.5, 60.7, 135.5,
          135.5 S318, 382.4, 243.2, 382.4z" id="textcircle" />
          <text class="heading">
              <textPath xlink:href="#textcircle">
                    ● LOADING ● LOADING ● LOADING
              </textPath>
          </text>
      </svg>
    </div>
  </div>
</template>

<style lang="postcss">
.loading-indicator {
  background: rgba(0,0,0,.9);
  @apply fixed top-0 left-0 w-full h-full flex items-center justify-center z-50;

  text {
    font-size: 2.5em;
    fill: #ecc94b;
    margin: 0;
    padding: 0;
    animation: spin 5000ms linear infinite;
    transform-box: fill-box;
    transform-origin: center;
    letter-spacing: -2px;
  }
}

@keyframes spin {
  0% {
    transform: rotate(360deg);
  },
  100% {
    transform: rotate(0);
  }
}
</style>
