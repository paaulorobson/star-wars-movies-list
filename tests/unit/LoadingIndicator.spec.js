import { shallowMount } from "@vue/test-utils"
import LoadingIndicator from "@/components/LoadingIndicator.vue"

test("Loading Indicator matches the snapshot", () => {
  const wrapper = shallowMount(LoadingIndicator)
  expect(wrapper.element).toMatchSnapshot()
})
