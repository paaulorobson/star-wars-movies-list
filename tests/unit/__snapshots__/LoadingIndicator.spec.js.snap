// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Indicator matches the snapshot 1`] = `
<div
  class="loading-indicator"
>
  <div
    class="loading-indicator__spinner"
  >
    <svg
      viewBox="0 0 500 500"
      width="280"
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
    >
      <title>
        LOADING
      </title>
       
      <path
        d="M243.2, 382.4c-74.8,
        0-135.5-60.7-135.5-135.5s60.7-135.5,135.5-135.5s135.5, 60.7, 135.5,
        135.5 S318, 382.4, 243.2, 382.4z"
        fill="transparent"
        id="textcircle"
      />
       
      <text
        class="heading"
      >
        <textpath
          xlink:href="#textcircle"
        >
          
                  ● LOADING ● LOADING ● LOADING
            
        </textpath>
      </text>
    </svg>
  </div>
</div>
`;
